File: Chapter 05/C#/Section_5_4_Server_Monitor/After/Infrastructure/Adapters/ConsoleAdapter.cs
using System;
using Chapter05.ServerMonitor.After.Core.Ports;

namespace Chapter05.ServerMonitor.After.Infrastructure.Adapters
{
    /// <summary>
    /// ADAPTER 2: The "Dev" Adapter.
    /// This adapter is perfect for local development. It proves to Archie that 
    /// the Core doesn't care if the alert goes to a multi-million dollar cloud 
    /// messaging service or simply prints to the local screen.
    /// </summary>
    public class ConsoleAdapter : IAlertPort
    {
        /// <summary>
        /// Implements the port by writing to the local console.
        /// </summary>
        public void SendAlert(string message)
        {
            // We use standard console colors to mimic a real alert,
            // but the Core logic remains completely unaware of this UI detail.
            Console.ForegroundColor = ConsoleColor.Red; 
            Console.WriteLine($"(DEV ADAPTER) ALERT: {message}"); 
            Console.ResetColor(); 
        }
    }
}

File: Chapter 05/Python/section_5_4_server_monitor/after/infrastructure/adapters/console_adapter.py
from core.ports.alert_port import AlertPort

class ConsoleAdapter(AlertPort):
    """
    ADAPTER 2: The 'Dev' Adapter.
    Proves to Archie that the Core doesn't care if the alert goes to a 
    cloud messaging service or simply prints to the local screen.
    """

    def send_alert(self, message: str) -> None:
        # We use ANSI escape codes to mimic a real red alert,
        # but the Core logic remains completely unaware of this UI detail.
        ansi_red = "\033[91m"
        ansi_reset = "\033[0m"
        print(f"{ansi_red}(DEV ADAPTER) ALERT: {message}{ansi_reset}")